File: topics/arrays/application/Max.pas
// Find the largest value in the array
function Max(const data: array of Double): Double;
var
  i: Integer;
begin
  result := data[Low(data)];
  for i := Low(data) + 1 to High(data) do
  begin
    if data[i] > result then result := data[i];
  end;
end;

File: topics/arrays/application/max.c
// Find the largest value in the array
double max(double data[], int size)
{
    int i;
    double max = data[0];
    
    for (i = 1; i < size; i++)
    {
        if (data[i] > max) max = data[i];
    }
    
    return max;
}
